src/lucky/routable.cr

```diff
@@ -145,14 +145,51 @@ module Lucky::Routable
     # no-op by default
   end
 
+  NORMALIZED_ROUTES = [] of Nil # types are not resolved in macros
+
+  # :nodoc:
+  macro enforce_route_uniqueness(method, original_path)
+    # Regex for capturing the param part for normalization
+    #
+    # So "/users/:user_id" is changed to "/users/:normalized"
+    {% normalized_path = original_path.gsub(/(\:\w*)/, ":normalized") %}
```

Comment:
This way `/users/:id` and `/users/:user_id` are counted as the same route, since that's how the router tries to match them.